### PR TITLE
use single containerd version for all k8s version

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 1]
```diff
@@ -127,7 +127,13 @@ cleanUpGPUDrivers() {
 installStandaloneContainerd() {
     UBUNTU_RELEASE=$(lsb_release -r -s)
     UBUNTU_CODENAME=$(lsb_release -c -s)
-    CONTAINERD_VERSION=$1
+    CONTAINERD_VERSION=$1    
+    # we always default to the .1 patch versons
+    CONTAINERD_PATCH_VERSION="${2:-1}"
+
+    # runc needs to be installed first or else existing vhd version causes conflict with containerd.
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:134]: this was a bug, runc sets CURRENT_VERSION too so it was always 1.0.3 and never using the cached version

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 2]
```diff
@@ -159,16 +159,17 @@ installStandaloneContainerd() {
...
+        CONTAINERD_PATCH_VERSION="4"
         echo "Containerd Version not specified, using default version: ${CONTAINERD_VERSION}-${CONTAINERD_PATCH_VERSION}"
     else
         echo "Using specified Containerd Version: ${CONTAINERD_VERSION}-${CONTAINERD_PATCH_VERSION}"
     fi
 
     CURRENT_MAJOR_MINOR="$(echo $CURRENT_VERSION | tr '.' '\n' | head -n 2 | paste -sd.)"
     DESIRED_MAJOR_MINOR="$(echo $CONTAINERD_VERSION | tr '.' '\n' | head -n 2 | paste -sd.)"
-    HAS_GREATER_VERSION="$(semverCompare "$CURRENT_VERSION" "$CONTAINERD_VERSION")"
+    semverCompare "$CURRENT_VERSION" "$CONTAINERD_VERSION"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:171]: this was a bug, it returns 0 but somehow the value of HAS_GREATER_VERSION was always "" (empty string) in the comparison against 0 below.

File: e2e/nodebootstrapping_template.json [hunk 3]
```diff
@@ -133,6 +133,7 @@
     "EnableGPUDevicePluginIfNeeded": false,
     "EnableKubeletConfigFile": false,
     "EnableNvidia": false,
+    "DisableUnattendedUpgrades": true,
```
Comment (alexeldeib) [inline e2e/nodebootstrapping_template.json:136]: will be relying on this in conjunction with https://github.com/Azure/AgentBaker/pull/2353/files to disable UU in e2e

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 4]
```diff
@@ -34,41 +34,22 @@ installContainerRuntime() {
...
     wait_for_file 120 1 /opt/azure/manifest.json # no exit on failure is deliberate, we fallback below.
 
-    local stable_containerd
-    local latest_containerd
+    local containerd_version
     if [ -f "$MANIFEST_FILEPATH" ]; then
-        stable_containerd="$(jq -r .containerd.stable "$MANIFEST_FILEPATH")"
-        latest_containerd="$(jq -r .containerd.latest "$MANIFEST_FILEPATH")"
-        edge_containerd="$(jq -r .containerd.edge "$MANIFEST_FILEPATH")"
+        containerd_version="$(jq -r .containerd.edge "$MANIFEST_FILEPATH")"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:39]: should really rename from edge -_-

will follow up when we remove this code entirely to tear it out. baby steps.

**Review by ganeshkumar5699 — COMMENTED** [review]
VHD installs the "latest" (1.5) version now and can be changed to install the "edge" version instead (1.6). Perhaps it was intended for another PR?

**Comment by alexeldeib** [comment]
> VHD installs the "latest" (1.5) version now and can be changed to install the "edge" version instead (1.6). Perhaps it was intended for another PR?

Nice catch. We should actually bump to 1.6 everywhere to avoid downgrading users who already have 1.6 I think.

**Comment by alexeldeib** [comment]
will update vhd install to account

**Review by ganeshkumar5699 — APPROVED** [review]

**Review by ganeshkumar5699 — APPROVED** [review]
Lgtm, besides some merge conflicts.

**Pull request merged**